### PR TITLE
Update workbench image recommended tag

### LINKED ISSUE #990: Revisit UI for OOTB images n & n-1 support
### Feature description

The intent is to give users the ability to select the latest, n,  and the previous version, n-1, for each OOTB image for notebooks.

Screenshot 2023-03-03 at 13 39 20
Screenshot 2023-03-03 at 13 39 15

### Describe alternatives you've considered

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by lucferbux**
## Description

Fix #990 

Ensure changes on OOTB images with support of n -1 version is working in the UI.

## How Has This Been Tested?

1. You need a second tag in the notebooks such as this [pr](https://github.com/atheo89/odh-manifests/commit/1046f453f7bbfc5bfd749e7c9af54a181aca3f59)
2. Add the `opendatahub.io/workbench-image-recommended: 'true'` tag to one of the notebook's images such as 's2i-minimal-notebook:py3.8-v1.3.1'
3. Check that the UI is displaying the recommended label
Screenshot 2023-03-06 at 11 57 22

Screenshot 2023-03-06 at 11 56 47

## Request review criteria:

- [X] The commits have meaningful messages (squashes happen on merge by the bot).
- [X] Included any necessary screenshots or gifs if it was a UI change.
- [X] Included tags to the UX team if it was a UI/UX change.
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has manually tested the changes and verified that the changes work
- [X] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

### CODE REVIEW

**Comment by lucferbux** [comment]
@kywalker-rh these are the changes for supporting n - 1 version, all seems correct such as the designs you provided.

**Comment by andrewballantyne** [comment]
You don't sort to the top -- @kywalker-rh is it more important that we keep the version order (from semver) or supersede it when we have a recommended version?

image

**Comment by kywalker-rh** [comment]
@andrewballantyne To me, recommended should always be at the top. Chances are that will usually coincide with the version order too, but recommended should be at the top and preselected by default.

**Review by andrewballantyne — CHANGES_REQUESTED** [review]
Yup, so one more item for you to address Lucas. Make sure the sort logic kicks recommended to the top.

**Comment by andrewballantyne** [comment]
/hold

Holding this until next release to sync up with the image updates as well.

**Comment by andrewballantyne** [comment]
/unhold

Needs another review -- but can merge now.

**Comment by lucferbux** [comment]
@andrewballantyne this can be merged now since we are targeting the new release

**Review by Gkrumbach07 — COMMENTED** [review]
code looks good. just had one question.

File: frontend/src/pages/projects/screens/spawner/spawnerUtils.ts
Comment (lucferbux) [reply frontend/src/pages/projects/screens/spawner/spawnerUtils.ts:115]: Yeah, we can add that.

Comment (lucferbux) [reply frontend/src/pages/projects/screens/spawner/spawnerUtils.ts]: Added a `const` since it's the same structure used in the backend, just to have them synced. I agree that this change improves legibility.

**Review by alexcreasy — COMMENTED** [review]
@lucferbux I've eyeballed the code and there's nothing that stands out. I'll run + test it once my cluster has spooled up.

**Review by Gkrumbach07 — APPROVED** [review]
/lgtm

**Review by alexcreasy — APPROVED** [review]
Code looks good to me.
Deployed and tested, the UI changes works as described in the PR.

**Review by andrewballantyne — APPROVED** [review]

**Comment by shalberd** [comment]
Very nice, enabling both dynamic and less dynamic image versions this way, selectable. This makes obsolete my concerns about GUI support. Plus, the fact that you only use non-SHA digest format in odh-manifests master branch imagestreams is good practice, too, purely from a development perspective.

**Pull request merged**
